frontend/src/features/research/activities/list/ActivityResults/AddActivityForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { PlusButton } from 'components/common/buttons';
+import { Select, SelectOption } from 'components/common/form';
+import { Formik } from 'formik';
+import { useApiResearchFile } from 'hooks/pims-api/useApiResearchFile';
+import useIsMounted from 'hooks/useIsMounted';
+import { Api_Activity, Api_ActivityTemplate } from 'models/api/Activity';
+import React, { useCallback, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+
+export interface IAddActivityFormProps {
+  onAddActivity: (activity: Api_Activity) => void;
+}
+
+export interface IAddActivity {
+  activityTypeId?: string;
+}
```

Comment:
is this necessary to abstract the Activityform data?

Reply:
not sure how to fix it, please suggest.

Reply:
My question is about the necessity of the interface given the name is not that descriptive. Is this interface necessary for the formik form? If so I would suggest renaming it

Reply:
Yes it is necessary formik form , changing it to IAddActivityForm as per convention in other form examples